File: 1.2 Patterns/12-triangle-with-num-3.py
row = 5
i = 0
j = 0
k = 0
l = 0
for i in range (0, row):
    for j in range (0, i + 1):
        print(j+1, end ="")
    for k in range (1, 2*(row - i ) - 1):
        print(" ", end ="")
    for l in range (j, -1, -1):
        print(l+1, end ="")
    print("")                           # \n is inserted by default

File: 1.2 Patterns/12-triangle-with-num-3.cpp
#include <iostream>
using namespace std;
int main()
{
    int row = 5, i, j, k;
    for( i = 0 ; i < row ; i++){
        for( j = 0 ; j <= i ; j++){
            cout<<j+1;
        }
        for( k = 1 ; k < 2*(row - i ) - 1 ; k++){
            cout<<" ";
        }
        for(;j > 0 ; j--){
            cout<<j;
        }
        cout<<"\n";
    }
    return 0;
}